File: src/test/java/com/gitlab/lae/intellij/actions/tree/WhenParsingTest.java
package com.gitlab.lae.intellij.actions.tree;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.junit.runners.Parameterized.Parameters;

import static java.util.Objects.requireNonNull;
import static org.junit.Assert.assertEquals;

@RunWith(Parameterized.class)
public final class WhenParsingTest {

    @Parameters(name = "{0}")
    public static Object[][] parameters() {
        return new Object[][]{
                {"ToolWindowActive:Project", When.toolWindowActive("Project")},
                {"ToolWindowTabActive:Log", When.toolWindowTabActive("Log")},
                {"FileExtension:rs", When.fileExtension("rs")},
                {"!FileExtension:rs", When.not(When.fileExtension("rs"))},
        };
    }

    private final String input;
    private final When expected;

    public WhenParsingTest(String input, When expected) {
        this.input = requireNonNull(input);
        this.expected = requireNonNull(expected);
    }

    @Test
    public void parsesInputCorrectly() {
        assertEquals(expected, When.parse(input));
    }
}

File: src/test/java/com/gitlab/lae/intellij/actions/tree/WhenParsingTest.kt
package com.gitlab.lae.intellij.actions.tree

import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.junit.runners.Parameterized.Parameters

@RunWith(Parameterized::class)
class WhenParsingTest(
  private val input: String,
  private val expected: When
) {

  @Test
  fun `parses input correctly`() {
    assertEquals(expected, When.parse(input))
  }

  companion object {
    @JvmStatic
    @Parameters(name = "{0}")
    fun parameters() = arrayOf(
      arrayOf("ToolWindowActive:Project", When.toolWindowActive("Project")),
      arrayOf("ToolWindowTabActive:Log", When.toolWindowTabActive("Log")),
      arrayOf("FileExtension:rs", When.fileExtension("rs")),
      arrayOf("!FileExtension:rs", When.not(When.fileExtension("rs")))
    )
  }
}
